src/mixins/window-width-mixin.js

```diff
@@ -0,0 +1,30 @@
+export default {
+  data: () => {
+    return {
+      winWidth: 0,
+    }
+  },
+  mounted() {
+    this.winWidth = window.innerWidth;
+    
+    // timeout reduces the frequency of resize events
+    let respondToResize;
+    window.onresize = () => {
```

Comment:
Instead of setting the `onresize` property, can you use `window.addEventListener('resize', ...);` instead. That way we can potentially have multiple event handlers listening on window resize event.

Reply:
Done.
It required some restructuring to facilitate removeEventListener when leaving the component.